utils.go

```diff
@@ -0,0 +1,52 @@
+package atomfs
+
+import (
+	"fmt"
+	"os"
+	"strings"
+)
+
+var TestOverrideRuntimeDirKey = "ATOMFS_TEST_RUN_DIR"
+
+func EnsureDir(dir string) error {
+	err := os.MkdirAll(dir, 0755)
+	if err != nil {
+		return fmt.Errorf("Failed creating directory %s: %w", dir, err)
+	}
+	return nil
+}
+
+func PathExists(d string) bool {
+	_, err := os.Stat(d)
+	if err != nil && os.IsNotExist(err) {
+		return false
+	}
+	return true
+}
+
+// remove dir separators to make one dir name. It is OK that this can't be
+// cleanly backed out, we don't need it to
+func ReplacePathSeparators(p string) string {
+	if p[0] == '/' {
+		p = p[1:]
+	}
+	return strings.ReplaceAll(p, "/", "-")
+}
+
+func GetMountNSName() (string, error) {
+	val, err := os.Readlink("/proc/self/ns/mnt")
+	if err != nil {
+		return "", err
+	}
+	// link target looks like 'mnt:[NUMBER]', we just want NUMBER
+	return val[5 : len(val)-1], nil
+}
+
+// Allow overriding runtime dir for tests so we can assert empty dirs, etc.
+func RuntimeDir() string {
+	testOverrideDir := os.Getenv(TestOverrideRuntimeDirKey)
```

Comment:
Hm, yeah good catch. Using this in the tests means I wasn't testing the case where the run dir was not writeable. :)

I guess a cli arg is the best option here.
It does mean that non-host-root usage will not actually have centralized metadata, so e.g. any scheme that tries to map dmverity faults to mounted dirs will fail to find the mount metadata.

Is there a better place to consolidate the metadata? What do you think about using /tmp/ instead of /run maybe?

Reply:
non-host-root cannot use dmverity, right?

Reply:
oh. yes, that's right. So the use cases don't overlap. OK. I am planning on adding a flag for this when I get a minute